strawberryfields/backends/states.py

```diff
@@ -832,6 +833,74 @@ def expand_dims(op, n, modes):
 
         return mean, var
 
+    def number_expectation(self, modes):
+        r"""
+        Calculates the expectation value of the product of the number operators of the modes.
+
+        This method computes the analytic expectation value
+        :math:`\langle \hat{n}_{i_0} \hat{n}_{i_1}\dots \hat{n}_{i_{N-1}}\rangle`
+        for a (sub)set of modes `[i_0, i_1, \dots, i_{N-1}]` of the system.
```

Comment:
(Needed for rendering purposes)
```suggestion
        for a (sub)set of modes :math:`[i_0, i_1, \dots, i_{N-1}]` of the system.
```